### DISCUSSION TITLE
Plugins: Enhance navigation

### DISCUSSION THREAD

**Discussion by dennisreimann** [btcpayserver/btcpayserver#3487] [General]
We have a "Plugins" drawer in the sidebar, which gives plugins an extension point to add items to the navigation. This works well for plugins with simple functionality, where the nav item links to a config view.

I would like to extend it so that plugins can have their own drawer, so that nav items can serve as more direct entry points: For instance in LNbank one can have several wallets. Right now one can access them by going to the index page that the single nav item points to. It would be a better experience if we could link directly to individual wallets from a LNbank sidebar drawer like this:   

![image](https://user-images.githubusercontent.com/886/154996208-a35466b5-b9ea-43c2-b980-ee5763bd7804.png)

So the proposal would be to offer an additional extension point for this kind of navigation.

Bonus request: We should rename the "Add Plugin" link to "Manage plugins", because this more appropriately describes the linked page.

**Comment by dstrukt** [comment]
ACK "Add Plugin" -> "Manage Plugins"

Introducing additional drawers for plug-ins is an interesting evolution .. I think it makes sense as btcpay scales to accommodate additional functionality and LNBank is a perfect test case for building out more plug-in infrastructure.

A few questions come to mind immediately:

1. Why would the view not look like the following to keep things encapsulated in their greater context .. I realize this may require a bit of design tinkering to really dial in:

-- PLUGINS
--- LNBANK
---- WORK
---- PERSONAL
--- PLUGIN
--- PLUGIN

I can see cases being made for either, just curious, and wanted to offer up that pattern for discussion/consideration.

2. Are there any privacy concerns w.r.t listing the balance in the always-visible left sidebar? What's the main reason for showing .. I imagine it's to make it easy to see you have a balance, as opposed to having to click into each instance to determine if there's any sats to sweep out...?

3. If we do keep the balance listed on the left, have you considered a shorter balance amount, e.g. 2,106,150 -> 2.1M sats or 15,180 -> 15.1K sats, up to maybe 3-4 trailing decimals? Might not be necessary to have the entire amount for a quick glance .. and would ultimately help reduce a bit of space .. but this is a visual nitpick.


Aside:

4. This is making me wonder if we should entertain the idea of adding the "+" to the right of the parent section as well as adjusting the down carot to potentially save a bit of vertical space.

Overall, I think these evolutions make sense, and it's just a matter of continuing to refine the designs to accommodate the changes. I think as we start to surface some at-a-glance widget to the dashboard this all becomes more exciting, even if there's more testing / overhead involved .. we'll eventually have to have some discussions around that & cross that bridge.

**Reply by dennisreimann** [reply]
As discussed in the design meeting, I solved it for now with this:

![grafik](https://user-images.githubusercontent.com/886/159085183-cfa94426-86f4-40a2-a5a3-02db6320f965.png)

**Reply by dstrukt** [reply]
Oh baby. 

Lookin' good!
